virtcontainers/hypervisor.go

```diff
@@ -574,6 +574,8 @@ type hypervisor interface {
 	addDevice(devInfo interface{}, devType deviceType) error
 	hotplugAddDevice(devInfo interface{}, devType deviceType) (interface{}, error)
 	hotplugRemoveDevice(devInfo interface{}, devType deviceType) (interface{}, error)
+	resizeMemory(memMB uint32, memoryBlockSizeMB uint32) (uint32, error)
+	resizeVCPUs(vcpus uint32) (uint32, uint32, error)
```

Comment:
These two overlap with `hotplugAddDevice` and `hotplugRemoveDevice`. Any reason those two cannot be used? I'm asking because if we keep the helper functions in sandbox.go, they can be reused by different hypervisor implementations.

Reply:
agreed that these should really be at sandbox level

Reply:
@bergwolf  the all idea is for the VM memory managment. For #793 will introduce manage the memory in with two methods balloon and hotplug. For longer term related with qemu implementation there is the option to use `virtio-mem` (`qmp` interface has a command called `memory-device-resize`).

 So when balloon is added (this PR is a depedency  of it), we think we dont want expose the balloon logic in the hypervisor interface or  hide it as part of hotplug/unplug memory calls. So the resizeMemory is looking to ask the hypervisor implementation manage the memory  as best as it could.  
Same for any other hypervisors that may have a variant of this methods (static memory, balloon, hotplug or something similar to virtio-mem), when we add them we will decide what is the best method for it but keep the hypervisor interface for vc simple. 

I think for resizeVCPUs() we could live without it but, wanted to give a similar method to the interface.  On vcpu managment only option today is hotplug.

Reply:
@bergwolf @sboeuf @egernst  is this point OK for you?

Reply:
@jcvenegas Yes I agree with you as this is something we did talk about offline.

@bergwolf @egernst the problem with `hotplug` methods is that they assume we will hotplug some CPUs or some memory. But it's really up to the hypervisor implementation to decide what he's gonna do when the sandbox require a specific amount of memory or CPUs.
As mentioned by @jcvenegas, this will allow for a clean abstraction, as the only thing that matters for the sandbox is to make sure it'll request what it needs. The implementation details of using hotplug or balloning, or anything else to perform what is asked, is simply left to the hypervisor implementation.

Reply:
Fair enough. I agree we can have a resource descriptive interface. Thanks @sboeuf @jcvenegas .